### PR TITLE
deleted editorOpenWith

### LINKED ISSUE #116272: Revisit editorOpenWith living on its own
I feel the `editorOpenWith.ts` thing needs to go away and move to respective services. It seems wrong to me that you pass in an editor group to operate on instead of having this exposed from the `IEditorGroupService`. Here is my thinking:
* add a new method to [`IEditorGroup`](https://github.com/microsoft/vscode/blob/630e706ad2de99e255a18bf77f72d569e9fc0e49/src/vs/workbench/services/editor/common/editorGroupsService.ts#L369-L369) (e.g. `openEditorWith`)
* clients will use `IEditorGroupService.getGroup().openEditorWith(...)`
* this method should also include the necessary dialog prompts for dealing with unsaved editors (from https://github.com/microsoft/vscode/pull/115264)

As for the remaining pieces of [`editorOpenWith.ts`](https://github.com/microsoft/vscode/blob/572bc1810dcb469c82f24d7c3135b2e12b72705e/src/vs/workbench/services/editor/common/editorOpenWith.ts#L200-L310) we can simply move them into `src/vs/workbench/common/editor.ts` at the end into a new folded region.

### PR DESCRIPTION
**Pull request by lramos15**
This PR fixes #116272.

I have moved the openWith function into the IEditorGroup and all the extra functions and helpers into editor.ts. I still need to add the unsaved file dialog. 

@bpasero I'm a bit confused as I feel parts of #116272 contradict #100821, specifically the portion about passing in an override option that shows the picker "we allow to pass in a special override option that will bring up a picker to pick a custom editor (this essentially moves the picker logic from openEditorWith into IEditorService". Would then openWith in the IEditorGroup not have any picker logic and instead be used just for opening a specific editor given an ID? Or would we say if there is no ID we pass that into the editorService to show the picker instead of the picker logic living within that function.

### CODE REVIEW

File: src/vs/workbench/common/editor.ts [hunk 1]
```diff
@@ -1637,3 +1642,147 @@ export function editorGroupToViewColumn(editorGroupService: IEditorGroupsService
...
+export const defaultEditorOverrideEntry = Object.freeze({
+	id: DEFAULT_EDITOR_ID,
+	label: localize('promptOpenWith.defaultEditor.displayName', "Text Editor"),
+	detail: builtinProviderDisplayName
+});
+
+/**
+ * Get a list of all available editors, including the default text editor.
+ */
+export function getAllAvailableEditors(
```
Comment (mjbvz) [inline src/vs/workbench/common/editor.ts:1681]: [nit] same for this function. I think it is only used in `openEditorWith`

**Review by mjbvz — APPROVED** [review]
Looks good to me overall but make sure @bpasero signs off too

**Review by bpasero — CHANGES_REQUESTED** [review]
Given a high level review feedback here before drilling into the code itself: I think `openEditorWith` should not be inside `editorGroupView` but rather be something I can pass to `IEditorService#openEditor` as an option. We already have the `override` property that I can pass to indicate which editor to use:

https://github.com/microsoft/vscode/blob/833bae4fd91e68edb2c93897be5146db3c8fa4cc/src/vs/workbench/common/editor.ts#L1012-L1012

In my mental model, an editor group should really not be involved with figuring out which editor to use for a given input. The editor group already deals with typed `EditorInput` that have an exact mapping to 1 editor. I would have rather expected the editor service to convert to the correct `EditorInput` (from a `{ resource }` untyped input) based on:
* `override`
* user settings (editor associations)
* extensions

**Comment by lramos15** [comment]
Closing in favor of https://github.com/microsoft/vscode/pull/116856 as I basically needed to restart so it was easier.

**Pull request closed without merging**